src/replication.c

```diff
@@ -1518,27 +1518,30 @@ static int useDisklessLoad() {
     return enabled;
 }
 
-/* Helper function for readSyncBulkPayload() to make backups of the current
- * databases before socket-loading the new ones. The backups may be restored
- * by disklessLoadRestoreBackup or freed by disklessLoadDiscardBackup later. */
-dbBackup *disklessLoadMakeBackup(void) {
-    return backupDb();
+/* Helper function for readSyncBulkPayload() to initialize tempDb
+ * before socket-loading the new db from master. The tempDb may be populated
+ * by swapDbWithTempDb or freed by disklessLoadDiscardTempDb later. */
+tempDb *disklessLoadInitTempDb(void) {
+    return initTempDb();
 }
 
-/* Helper function for readSyncBulkPayload(): when replica-side diskless
- * database loading is used, Redis makes a backup of the existing databases
- * before loading the new ones from the socket.
- *
- * If the socket loading went wrong, we want to restore the old backups
- * into the server databases. */
-void disklessLoadRestoreBackup(dbBackup *backup) {
-    restoreDbBackup(backup);
+/* Helper function for readSyncBulkPayload() to discard our tempDb
+ * when the loading succeeded. */
+void disklessLoadDiscardTempDb(tempDb *tempDb) {
+    discardTempDb(tempDb, replicationEmptyDbCallback);
 }
 
-/* Helper function for readSyncBulkPayload() to discard our old backups
- * when the loading succeeded. */
-void disklessLoadDiscardBackup(dbBackup *backup, int flag) {
-    discardDbBackup(backup, flag, replicationEmptyDbCallback);
+void swapDbWithTempDb(tempDb* tempDb) {
```

Comment:
I think this method belongs in db.c, maybe we want a thin wrapper for it here, or maybe we wanna give up all the thin wrappers (their main purpose was their doc comments i think)
```suggestion
/* Logically, this discards (flush) the old main database, and apply the newly loaded
 * database (temp) as the main (active) database, the actual freeing of old database
  * (which will now be placed in the temp one) is done later. */
void swapMainDbWithTempDb(tempDb* tempDb) {
```

Reply:
Added your method comment, renamed to swapMainDbWithTempDb, moved to db.c (merging with dbSwapAllDatabases), removed thin wrapper from replication.c.